e2e/test_operator/operator_test.go

```diff
@@ -124,6 +126,265 @@ var _ = Describe("Operator", Label("e2e"), func() {
 		}
 	})
 
+	PContext("testing taint feature", func() {
+		taintKeyMaintenance := "maintenance"
+		taintKeyMaintenanceDuration := int64(2)
+		taintKeyStar := "*"
+		taintKeyStarDuration := int64(5)
+		var initialPods *corev1.PodList
+		var taintedNode *corev1.Node
+		var taintedNodes []*corev1.Node
+		var historyTaintedNodes map[string]bool
+		numNodesTainted := 2 // TODO: Change to higher number 5
+		ensurePodIsDeletedTimeoutMinutes := 20
+
+		BeforeEach(func() {
+			// Cleanup tainted nodes in case previous test fails which will leave tainted nodes behind
+			for nodeName := range historyTaintedNodes {
```

Comment:
In case a previous test, which taints several nodes, fails, it will leave those tainted nodes as tainted.  
If we don't untaint those tests, the next test may likely fail because it is hard, and more often impossible, to find a pod on not-tainted node. That makes the test stuck in reconciliation and fail incorrectly.

All in all, this is a hack to mitigate the sad situation that the test is running in a shared k8s cluster. Later if we have kind based testing, we can just untaint all nodes or simply check all nodes are not-tainted.

Reply:
As to historyTaintedNodes  instead of `taintedNodes `,  I'd like to ensure the all nodes are untainted. 
If `err := fdbCluster.getClient().Update(ctx.Background(), node)` fails in previous test's AfterEach, the node is left tainted. (I have seen this a lot. That's why I added the Eventually in the [UpdateNode](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/05f282b84b022ec18d866ba30b1e5816a345e697/e2e/fixtures/factory.go#L769).

Anyway, I don't like this approach either. I would rather simply untaint all nodes.